Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -4029,73 +4102,189 @@ public ObservableList<ElementLocator> GetElementFriendlyLocators(ElementInfo Ele
 
         public string GetAppPackageNameByOs()
         {
-            return Driver is AndroidDriver ? "package" : "bundleId";
+            try
+            {
+                if (Driver is AndroidDriver)
+                {
+                    return "package";
+                }
+                else if (Driver is IOSDriver)
+                {
+                    return "bundleId";
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Unsupported driver type: {Driver.GetType().Name}");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new($"Failed to get app identifier attribute name: {ex.Message}");
```

Comment:
mention type of exception as "InvalidOperationException" in all exceptions without type added in class